internal/graph/graph.go

```diff
@@ -74,3 +74,50 @@ func max(x, y uint32) uint32 {
 	}
 	return x
 }
+
+var emptyMetadata = &v1.ResponseMeta{}
+
+func updateRespMetadata(existing *v1.ResponseMeta, responseMetadata *v1.ResponseMeta) *v1.ResponseMeta {
```

Comment:
Should we rename this to combineResponseMetadata? I assumed it updated the first parameter in place. We could also make it take a list, collect up any that need to be combined, and combine them all lazily, that would create less ResponseMeta garbage for `any` and `all` calls.

Reply:
The downside of taking a list is that it makes it harder to pass around; we could create an intermediate struct, add the responses into that, and then combine as a final step. I'm all for renaming this though, as it is somewhat vague

Reply:
Updated the name